From fef254ab94691e36a7b75072bec76caf68169d39 Mon Sep 17 00:00:00 2001
From: Mauro Carvalho Chehab <m.chehab@samsung.com>
Date: Sat, 24 Aug 2013 08:47:51 -0300
Subject: [media] vsp1: Fix a sparse warning

As reported by: kbuild test robot <fengguang.wu@intel.com>:
	 drivers/media/platform/vsp1/vsp1_drv.c:434:21: sparse: cast removes address space of expression

	   433		vsp1->mmio = devm_ioremap_resource(&pdev->dev, io);
	 > 434		if (IS_ERR((void *)vsp1->mmio))
	 > 435			return PTR_ERR((void *)vsp1->mmio);

There's no need to convert it to void *.

Reported-by: kbuild test robot <fengguang.wu@intel.com>
Signed-off-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
(cherry picked from commit cbec6d3ab470565536480d3bd109a7fdb128c3c4)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 drivers/media/platform/vsp1/vsp1_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/vsp1/vsp1_drv.c b/drivers/media/platform/vsp1/vsp1_drv.c
index 8700842edcf2..ff8cd2d6be39 100644
--- a/drivers/media/platform/vsp1/vsp1_drv.c
+++ b/drivers/media/platform/vsp1/vsp1_drv.c
@@ -434,8 +434,8 @@ static int vsp1_probe(struct platform_device *pdev)
 	/* I/O, IRQ and clock resources */
 	io = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	vsp1->mmio = devm_ioremap_resource(&pdev->dev, io);
-	if (IS_ERR((void *)vsp1->mmio))
-		return PTR_ERR((void *)vsp1->mmio);
+	if (IS_ERR(vsp1->mmio))
+		return PTR_ERR(vsp1->mmio);
 
 	vsp1->clock = devm_clk_get(&pdev->dev, NULL);
 	if (IS_ERR(vsp1->clock)) {
-- 
1.8.5.rc3

